src/puppet.ts

```diff
@@ -853,11 +853,19 @@ export abstract class Puppet extends EventEmitter {
       return allRoomIdList
     }
 
-    const roomPayloadList = await Promise.all(
+    const roomPayloadList: RoomPayload[] = (await Promise.all(
       allRoomIdList.map(
-        id => this.roomPayload(id),
+        async id => {
+          try {
+            return await this.roomPayload(id)
+          } catch (e) {
+            // compatible with {} payload
```

Comment:
If the room id is invalid, should we delete them from the cache so that we will not get the non-exist payload of that id again?

If so, I think the following code will get it done:

```ts
await this.roomPayloadDirty(id)
await this.roomMemberPayloadDirty(id)
```

Reply:
Glad to know that you can fix more, thank you very much.

Reply:
Just updated my comment above, please take a look.

Reply:
Yes, I agree with you that the low-level data should be maintained by the puppet implementation, it's the responsibility of the puppet implementation,  and they should do well.

At the same time, I believe it's also necessary to clean the invalid data in puppet base class instead of leaving as it is.

Reply:
Okay, I got it, will update the PR soon.